app/models/miq_worker/runner.rb

```diff
@@ -332,7 +332,10 @@ def do_work_loop
       begin
         heartbeat
         do_work
-      rescue TemporaryFailure
+      rescue TemporaryFailure => error
+        msg = "#{log_prefix} Temporary failure (message: '#{error}') caught"\
+            " during #do_work. Sleeping for a while before resuming."
+        _log.warn(msg)
```

Comment:
@durandom I think it makes sense to log this in the main log since it's worker related.

Should this follow a similar pattern to authentication check at boot where we won't start if the creds are bad or if they system is temporarily not available?  Like [this](https://github.com/ManageIQ/manageiq/blob/f4b27ed2ec355afc4d3570abe7a260af0a0e3992/app/models/manageiq/providers/base_manager/event_catcher/runner.rb#L17)